webpack.config.js

```diff
@@ -2,53 +2,74 @@ const path = require('path');
+const TerserPlugin = require('terser-webpack-plugin');
+const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
 
-module.exports = {
-  entry: './src/index.tsx',
-  resolve: { extensions: ['.ts', '.tsx', '.js', '.jsx'] },
-  output: {
-    filename: 'bundle.js',
-    path: path.join(__dirname, '/dist'),
-    clean: true
-  },
-  devServer: {
-    hot: true,
-    open: true,
-    historyApiFallback: true
-  },
-  devtool: 'source-map',
-  plugins: [
-    new HtmlWebpackPlugin({
-      template: './index.html'
-    }),
-    new CopyWebpackPlugin({
-      patterns: [{ from: './public', to: './public' }]
-    }),
-    new Dotenv()
-  ],
-  module: {
-    rules: [
-      {
-        test: /\.(js|jsx|ts|tsx)$/i,
-        exclude: /node_modules/,
-        use: {
-          loader: 'ts-loader'
-        }
-      },
-      {
-        test: /\.css$/i,
-        use: ['style-loader', 'css-loader']
-      },
-      {
-        test: /\.(eot|svg|ttf|woff|woff2|png|jpg|gif)$/i,
-        loader: 'file-loader',
-        options: {
-          name: 'static/[name].[ext]'
+module.exports = (env, argv) => {
+  const isDevelopment = argv.mode === 'development';
+
+  return {
+    entry: './src/index.tsx',
+    resolve: { extensions: ['.ts', '.tsx', '.js', '.jsx'] },
+    output: {
+      filename: 'bundle.[chunkhash].js',
+      path: path.join(__dirname, '/dist'),
+      clean: true
+    },
+    devServer: {
+      hot: true,
+      open: true,
+      historyApiFallback: true
+    },
+    devtool: 'source-map',
+    plugins: [
+      new HtmlWebpackPlugin({
+        template: './index.html'
+      }),
+      new CopyWebpackPlugin({
+        patterns: [{ from: './public', to: './public' }]
+      }),
+      new Dotenv(),
+      new MiniCssExtractPlugin(),
+      new BundleAnalyzerPlugin({
+        analyzerMode: 'static',
+        openAnalyzer: false,
+        generateStatsFile: true,
+        statsFilename: 'bundle-report.json'
+      })
+    ],
+    module: {
+      rules: [
+        {
+          test: /\.(js|jsx|ts|tsx)$/i,
+          exclude: /node_modules/,
+          use: {
+            loader: 'ts-loader'
+          }
+        },
+        {
+          test: /\.css$/i,
+          use: [isDevelopment ? 'style-loader' : MiniCssExtractPlugin.loader, 'css-loader']
+        },
+        {
+          test: /\.(eot|svg|ttf|woff|woff2|png|jpg|gif|webp|mp4|avif)$/i,
+          loader: 'file-loader',
+          options: {
+            name: 'static/[name].[ext]'
+          }
+        },
+        {
+          test: /\.(png|jpg|jpeg|gif|webp|mp4)$/i,
+          loader: 'image-webpack-loader',
+          enforce: 'pre'
         }
-      }
-    ]
-  },
-  optimization: {
-    minimize: false
-  }
+      ]
+    },
+    optimization: {
+      splitChunks: { chunks: 'all' },
+      minimizer: [new TerserPlugin(), new CssMinimizerPlugin()]
```

Comment:
```suggestion
      minimizer: [`...`, new CssMinimizerPlugin()]
```

[웹팩5](https://webpack.kr/plugins/json-minimizer-webpack-plugin/#root)에서 Production 모드일 때 TerserPlugin으로 자동으로 최적화를 해주기 때문에 위와 같이 수정하면 좋을 것 같아요~~

Reply:
앗 감사합니다~!!수정했어요!!